src/pudl/transform/ferc1.py

```diff
@@ -54,6 +54,37 @@ def clean_xbrl_metadata_json(
     return add_source_tables_to_xbrl_metadata(raw_xbrl_metadata_json)
 
 
+def all_fields_in_table(table_meta) -> list:
+    """Compile a list of all of the fields reported in a table."""
+    return [field["name"] for meta_list in table_meta.values() for field in meta_list]
+
+
+def extract_tables_to_fields(raw_xbrl_metadata_json: dict) -> dict[str, list[str]]:
+    """Compile a dictionary of table names (keys) to list of fields.
+
+    Args:
+        raw_xbrl_metadata_json: dictionary of xbrl metadata.
+    """
+    return {
+        table_name: all_fields_in_table(table_meta)
+        for table_name, table_meta in raw_xbrl_metadata_json.items()
+    }
+
+
+def label_source_tables(calc_component: dict, tables_to_fields: str) -> dict:
+    """Add a ``source_tables`` element to the calculation component."""
+    calc_component["source_tables"] = [
+        table_name
+        for table_name, fields in tables_to_fields.items()
+        if calc_component["name"] in fields
+    ]
+    # weirdly there are a number of nuclear xbrl_factoid calc components that seem
+    # to have no source_tables.
+    if not calc_component["source_tables"]:
+        logger.debug(f"Found no source table for {calc_component['name']}.")
+    return calc_component
```

Comment:
This seems fishy. Is there a rename issue somewhere maybe? Do they not show up in the XBRL taxonomy anywhere?

Reply:
these have not been renamed at all at this point. and yes to the latter that definitely happens sometimes. as in there are references in the calcs to fields that do not exist i just moved these little internal functions out of the main label function, i didn't edit them at all.